ui/src/serverIO.js

```diff
@@ -376,27 +376,24 @@ class ServerIO {
       this.dispatch(showResumeQueueDialog(true));
     });
 
-    this.hwrSocket.on('observersChanged', (data) => {
-      const state = store.getState();
+    this.hwrSocket.on('userChanged', async (message) => {
+      const prevState = store.getState();
+      const { inControl: wasInControl } = prevState.login.user;
 
-      if (
-        data.observers.length > 0 &&
-        data.operator.username === state.login.user.username &&
-        !state.login.user.inControl
-      ) {
-        this.dispatch(showWaitDialog('You were given control', data.message));
-      } else if (
-        data.observers.length > 0 &&
-        state.login.user.inControl &&
-        data.observers
-          .map((el) => el.username)
-          .includes(state.login.user.username)
-      ) {
+      await this.dispatch(getLoginInfo());
```

Comment:
So each message now refecthes its own slice of state:

- `userChanged` dispatches `getLoginIngo()`
- `observersChanged` dispatches `getRaState()`